src/PerfView/PerfViewData.cs

```diff
@@ -5729,10 +5732,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                         allocs = GetHeap(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
                                     }
 
+                                    if (allocationSites == null)
+                                    {
+                                        var symReader = GetSymbolReader(log, SymbolReaderOptions.CacheOnly);
+                                        allocationSites = new Dictionary<TraceModuleFile, List<HeapAllocationSite>>();
+                                        foreach (var item in data.Process().LoadedModules)
+                                        {
+                                            if (symReader.FindSymbolFilePathForModule(item.FilePath) is string pdb &&
+                                                symReader.OpenNativeSymbolFile(pdb) is NativeSymbolModule symbolFile)
+                                            {
+                                                var sites = symbolFile.GetHeapAllocationSites().ToList();
```

Comment:
Consider:   Maybe you just have NativeSymbolModule implement GetTypeForHeapAllocationSite(uint rva), (and a bool HasHeapAllocationSiteInfo)   Then you don't need to enumerate the tuples.  Either you can use the DIA interface to look up the allocation sites by RVA (it probably offers that, or at worse, you make a NativeSymbolModule hold the RVA->string dictionary.    That avoids even needing to define the HeapAllocation struct (which is just a copy anyway).